src/cleanupCommand/cleanupManager.ts

```diff
@@ -39,6 +43,16 @@ export class CleanupManager {
       const completedDiscretes = expiredBatch.filter((el) => !failedDiscreteLayers.includes(el));
       await this.jobManager.markAsCompletedAndRemoveFiles(completedDiscretes);
     }
+
+    for (let i = 0; i < notCleanedAndFailedUpdate.length; i += this.discreteBatchSize) {
+      const currentBatch = notCleanedAndFailedUpdate.slice(i, i + this.discreteBatchSize);
+      const expiredBatch = this.filterExpiredFailedTasks(currentBatch, deleteDate);
+      if (expiredBatch.length > 0) {
+        await this.sourcesProvider.deleteDiscretes(expiredBatch);
+      }
+
+      await this.jobManager.markAsCompletedAndRemoveFiles(notCleanedAndFailedUpdate);
```

Comment:
await this.jobManager.markAsCompletedAndRemoveFiles(expiredBatch);

Reply:
done.